libbeat/docs/security/users.asciidoc

```diff
@@ -0,0 +1,295 @@
+If you use the +{beat_default_index_prefix}_system+ user, make sure you
+<<beats-system-user,set the password>>.
+
+[options="header"]
+|====
+|Task | Required privileges
+
+|Send monitoring info | `monitor` on cluster
+|====
+
+// REVIEWERS: I don't have time to test the full monitoring set up. Is this
+// correct, or are additional privileges required?
+
+
+[[privileges-to-publish-events]]
+==== Privileges needed to publish events
+
+Users who publish events to {es} need to create and read from {beatname_uc}
+indices. The privileges required for this role depend on the tasks users
+need to perform:
+
+[options="header"]
+|====
+|Task | Required privileges
+
+.2+|Send data to a secured cluster
+|`monitor` on cluster
+
+|`create_index` and `index` on {beatname_uc} indices
+
+.2+|Send data to a secured cluster when automatic template loading is enabled
+footnote:[Automatic template loading is enabled by
+default. To disable it, set `setup.template.enabled: false`. Note that you
+cannot disable automatic template loading if you're using index lifecycle
+management.]
+
+|`monitor` and `manage_index_templates` on cluster
+
+|`create_index` and `index` on {beatname_uc} indices
+
+ifeval::["{beatname_lc}"=="filebeat"]
+.2+|Send data to a secured cluster when {beatname_uc} modules are enabled
+|`monitor` and `manage_ingest_pipelines` on cluster
+
+|`create_index` and `index` on {beatname_uc} indices
+endif::[]
+
+.2+|Send data to a secured cluster when index lifecycle management is enabled
+|`manage_index_templates` and `monitor` on cluster
+
+| `index` and `manage` on {beatname_uc} indices
+ifdef::has_central_config[]
+.2+|Read configurations from Beats central management
+| `monitor` on cluster
+
+|`all` on `.management-beats` index
+endif::has_central_config[]
+|====
+
+//REVIEWERS: Some questions about the above info:
+//
+// 1 - When is `read_ilm` needed?
+// Also, do users need `index` and `manage` on `shrink-beatname-*`, too?
+// Are there other privileges that might be required as indices move through the
+// lifecycle stages?
+//
+// 2 - Is there a more restrictive privilege than `all` that we can assign to users who
+// need to read configs in central management, but not craete or manage configs? I tried
+// `read` on .management-beats but got "ERROR process/process.go:47 Error getting
+// process details. pid=35087: error getting process mem for pid=35087:
+// Could not read process info for pid 35087" Beats still sent events.
+
+// REVIEWERS: I have some concerns about how our default config settings affect
+// the privileges users need to assign:
+// In some cases, users either need to change the defaults that we've set, or
+// grant privileges that are less restrictive than they might want.
+// For example, on 6.7, I get the following
+// error when I run Filebeat (with the system module enabled)
+// when the user doesn't have the `manage_ingest_pipelines` cluster
+// privilege (this happens even if I load the pipeline in advance):
```

Comment:
Yeah, this is why we should encourage users to run `filebeat setup` with another user.

Reply:
I did this, but the privilege was still required. I will retest to see if I can reproduce the behavior in 7.0 as well as 6.7.